src/status_im/transport/filters.cljs

```diff
@@ -20,43 +23,55 @@
                      #(log/warn :add-filter-error (.stringify js/JSON (clj->js options)) %)))
 
 (defn add-filter!
-  [web3 {:keys [topics to] :as options} callback]
+  [web3 {:keys [chat-id topics to event] :as options} callback]
   (let [options  (if config/offline-inbox-enabled?
                    (assoc options :allowP2P true)
                    options)]
     (log/debug :add-filter options)
-    (add-shh-filter! web3 options callback)))
+    (if-let [filter (add-shh-filter! web3 options callback)]
```

Comment:
Why can't this be done in callback?

Reply:
I was wondering the same, but decided not to change it to keep code changes to a minimum. The callback version would take 4 arguments https://web3js.readthedocs.io/en/1.0/web3-shh.html#shh-subscribeoptions (subscribe has the same signature as  newmessagefilter), with the last being the returned value, I can try and see if it works.

Reply:
no dice, callback does not fire on success, only on error

Reply:
@rasom could you have another look at the PR? I had to rebase against the linting changes and I don't want to be the first one to introduce badly formatted code :)

Reply:
@cammellos One area where I found your formatting not following the clojure styleguide are the comments, you use `;` comment prefix for standalone comment lines, which is wrong (`;` is only to be used for after the line comments, which was agreed on not to use), `;;` should be used instead.
Probably not related to this PR, more like an general remark.

Reply:
@janherich guilty as charged, I didn't know the difference, thanks!